ansible/playbooks/aix.yml

```diff
@@ -399,78 +399,132 @@
-          tags: xlc
+          when: xlc13.stat.islnk is not defined
+          tags: xlc13
 
         - name: Symlink - /usr/bin/xlc++
           file: src=/opt/IBM/xlC/13.1.3/bin/xlc++ dest=/usr/bin/xlc++ state=link
-          when: xlc.stat.islnk is not defined
-          tags: xlc
+          when: xlc13.stat.islnk is not defined
+          tags: xlc13
 
         - name: Symlink - /usr/bin/xlC
           file: src=/opt/IBM/xlC/13.1.3/bin/xlC dest=/usr/bin/xlC state=link
-          when: xlc.stat.islnk is not defined
-          tags: xlc
+          when: xlc13.stat.islnk is not defined
+          tags: xlc13
 
         - name: Symlink - /usr/bin/xlc_r
           file: src=/opt/IBM/xlC/13.1.3/bin/xlc_r dest=/usr/bin/xlc_r state=link
-          when: xlc.stat.islnk is not defined
-          tags: xlc
+          when: xlc13.stat.islnk is not defined
+          tags: xlc13
 
         - name: Symlink - /usr/bin/xlC_r
           file: src=/opt/IBM/xlC/13.1.3/bin/xlc_r dest=/usr/bin/xlC_r state=link
-          when: xlc.stat.islnk is not defined
-          tags: xlc
+          when: xlc13.stat.islnk is not defined
+          tags: xlc13
 
         - name: Symlink - /usr/bin/gxlC
           file: src=/opt/IBM/xlC/13.1.3/bin/gxlC dest=/usr/bin/gxlC state=link
-          when: xlc.stat.islnk is not defined
-          tags: xlc
+          when: xlc13.stat.islnk is not defined
+          tags: xlc13
+
+        - name: Clean XLC13 tmp files
+          file:
+            path: /tmp/usr
+            state: absent
+          tags: xlc13
+
+      ###################
+      # IBM XL C 16.1.0 #
+      ###################
+        - name: Checking for XLC16 availability
+          stat:
+            path: /opt/IBM/xlc/16.1.0/bin/xlc
+          register: xlc16
+          tags: xlc16
+
+        - debug:
+            msg: "xlc16 installed, skipping download and installation"
+          when: xlc16.stat.islnk is defined
+          tags: xlc16
+
+        - name: Transfer and Extract XLC16
+          unarchive:
+            src: /Vendor_Files/aix/XLC/XL_C_Cpp_FOR_AIX_V16.1_EMG.tar.Z
+            dest: /tmp
+            remote_src: no
+          when: xlc16.stat.islnk is not defined
+          tags: xlc16
+
+        - name: Install IBM XLC16 - installp
+          command: installp -aXYgd /tmp/usr/sys/inst.images all
+          register: result.xlc
+          ignore_errors: yes
+          when: xlc16.stat.islnk is not defined
+          tags: xlc16
+
+        - debug: msg='Erorrs from the previous installp command normal'
+          when: xlc16.stat.islnk is not defined
+          tags: xlc16
+
+        - name: TestIBM XLC16
+          command: /opt/IBM/xlC/16.1.0/bin/xlc -qversion
+          register: xlc16_qversion
+          tags: xlc16
+
+        - name: Display XLC16 qversion information
+          debug:
+            msg: "{{ xlc16_qversion.stdout }}"
+          tags: xlc16
+
+        - name: Clean XLC16 tmp files
+          file:
```

Comment:
There is also a log file that needs to be deleted: /tmp/install.log
You could also add `-e /tmp/usr/install.log` to the`installp ...` command above, thus no extra clean up required

Reply:
Good idea; updated both cases.  Is this how you mean?

Reply:
see https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/773/files#r349738526